cmd/manager/main.go

```diff
@@ -147,64 +159,72 @@ func main() {
 	if clusterOperatorName != "" {
 		statusReporter, err = status.NewReporter(cfg, mgr, namespace, clusterOperatorName, os.Getenv("RELEASE_VERSION"), stopCh)
 		if err != nil {
-			exit(err)
+			logrus.Error(err)
+			os.Exit(1)
 		}
 	}
 
 	// Populate the global default OperatorSources definition and config
 	err = defaults.PopulateGlobals()
 	if err != nil {
-		exit(err)
+		logrus.Error(err)
+		os.Exit(1)
 	}
 
 	// Setup all Controllers
 	if err := controller.AddToManager(mgr, options.ControllerOptions{}); err != nil {
-		exit(err)
+		logrus.Error(err)
+		os.Exit(1)
 	}
 
-	// Serve a health check.
-	http.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
-		w.WriteHeader(http.StatusOK)
-	})
-	go http.ListenAndServe(":8080", nil)
+	logrus.Info("Setting up health checks")
+	if err := mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
+		logrus.Error(err, "unable to set up health check")
+		os.Exit(1)
+	}
+	if err := mgr.AddReadyzCheck("healthz", healthz.Ping); err != nil {
+		logrus.Error(err, "unable to set up ready check")
+		os.Exit(1)
+	}
 
 	// Wait until this instance becomes the leader.
-	log.Info("Waiting to become leader.")
-	err = leader.Become(context.TODO(), "marketplace-operator-lock")
+	logrus.Info("Waiting to become leader.")
+	err = leader.Become(context.TODO(), leaderElectionConfigMapName)
```

Comment:
Unlikely to change, but I moved the name of the leader election ConfigMap to a package-level constant to avoid hardcoding this value. It's worth re-evaluating whether we want to continue using the leader-for-life implementation, like we're using here, vs. the controller-runtime leader-for-lease implementation that's configurable in the manager instance.